vividus-engine/src/main/java/org/vividus/annotation/ReplacementMonitor.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2019-2023 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.annotation;
+
+import java.lang.annotation.Annotation;
+import java.lang.reflect.Method;
+import java.util.Formatter;
+import java.util.Optional;
+import java.util.regex.Matcher;
+
+import org.jbehave.core.annotations.Alias;
+import org.jbehave.core.annotations.Given;
+import org.jbehave.core.annotations.Then;
+import org.jbehave.core.annotations.When;
+import org.jbehave.core.configuration.Configuration;
+import org.jbehave.core.configuration.Keywords;
+import org.jbehave.core.parsers.StepMatcher;
+import org.jbehave.core.steps.NullStepMonitor;
+import org.jbehave.core.steps.StepType;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.vividus.context.RunTestContext;
+
+public class ReplacementMonitor extends NullStepMonitor
+{
+    private static final Logger LOGGER = LoggerFactory.getLogger(ReplacementMonitor.class);
+
+    private final Configuration configuration;
+    private final RunTestContext runTestContext;
+
+    public ReplacementMonitor(Configuration configuration, RunTestContext runTestContext)
```

Comment:
inject particular beans instead of `Configuration` to avoid circular dependencies

Reply:
> inject particular beans instead of `Configuration` to avoid circular dependencies

We need from configuration `Keywords` and `StepPatternParser`, and both not beans